pkg/operator/controller/garden/garden/components.go

```diff
@@ -1036,6 +1042,26 @@ func (r *Reconciler) newGardenerScheduler(garden *operatorv1alpha1.Garden, secre
 	return gardenerscheduler.New(r.RuntimeClientSet.Client(), r.GardenNamespace, secretsManager, values), nil
 }
 
+func (r *Reconciler) newCertManagement(garden *operatorv1alpha1.Garden) (component.DeployWaiter, error) {
+	config := garden.Spec.RuntimeCluster.CertManagement
+	if config == nil {
+		return certmanagement.NewDestroyer(certmanagement.New(r.RuntimeClientSet.Client(), certmanagement.Values{})), nil
+	}
```

Comment:
Other components in the same controller handle the same case with an own deploy function ([ref](https://github.com/gardener/gardener/blob/d30f7f5a57a86dc970d4700b4c62e79590a01fcc/pkg/operator/controller/garden/garden/reconciler_reconcile.go#L291-L296), [ref](https://github.com/gardener/gardener/blob/d30f7f5a57a86dc970d4700b4c62e79590a01fcc/pkg/operator/controller/garden/garden/reconciler_reconcile.go#L701-L703)) instead of an own Destroyer component.
Please do it the same way to keep things consistent.

Reply:
refactored with multiple deployers